go-controller/pkg/ovn/egressqos.go

```diff
@@ -786,6 +786,18 @@ func (oc *DefaultNetworkController) onEgressQoSPodAdd(obj interface{}) {
 		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %+v: %v", obj, err))
 		return
 	}
+	pod := obj.(*kapi.Pod)
+	// only process this pod if it is local to this zone
+	if !oc.isPodScheduledinLocalZone(pod) {
+		// NOTE: This means we don't handle the case where pod goes from
+		// being local to remote. So far there is no use case for this to happen.
```

Comment:
This does look a bit weird as in other places we are handling this although I don't have a problem if we decide not to support it here specifically.

But this does potentially leave the system in an inconsistent state (OVN status is not reconciled with respect k8s status) and while we might choose not to support this gracefully, understood as the functionality itself might not work without disruption, leaving the DB in an inconsistent state does make my hair stand up a bit.

I guess we do handle pods becoming local, even if after they had been scheduled.

Would it be reasonable and relatively easy to process pods that have gone remote as if they had been deleted?

Reply:
I guess that would entail also processing node changes, to resync all the node's pod if it changes zone.

Reply:
i thought about this case.. for this controller i am convinced we don't need to do anything here.
In all scenarios of pod and node zone updates for the local->remote the addUpdateNodeRemote path will just delete the switch totally.. that means all pods will be drained and all the rules are also going to be deleted/garbage collected i guess?
```
"Logical_Switch": {
            "columns": {
                "name": {"type": "string"},
                "ports": {"type": {"key": {"type": "uuid",
                                           "refTable": "Logical_Switch_Port",
                                           "refType": "strong"},
                                   "min": 0,
                                   "max": "unlimited"}},
                "acls": {"type": {"key": {"type": "uuid",
                                          "refTable": "ACL",
                                          "refType": "strong"},
                                  "min": 0,
                                  "max": "unlimited"}},
                "qos_rules": {"type": {"key": {"type": "uuid",
                                          "refTable": "QoS",
                                          "refType": "strong"},
                                  "min": 0,
                                  "max": "unlimited"}},
```
```
        "QoS": {
            "columns": {
                "priority": {"type": {"key": {"type": "integer",
                                              "minInteger": 0,
                                              "maxInteger": 32767}}},
                "direction": {"type": {"key": {"type": "string",
                                            "enum": ["set", ["from-lport", "to-lport"]]}}},
                "match": {"type": "string"},
                "action": {"type": {"key": {"type": "string",
                                            "enum": ["set", ["dscp"]]},
                                    "value": {"type": "integer",
                                              "minInteger": 0,
                                              "maxInteger": 63},
                                    "min": 0, "max": "unlimited"}},
                "bandwidth": {"type": {"key": {"type": "string",
                                               "enum": ["set", ["rate",
                                                                "burst"]]},
                                       "value": {"type": "integer",
                                                 "minInteger": 1,
                                                 "maxInteger": 4294967295},
                                       "min": 0, "max": "unlimited"}},
                "external_ids": {
                    "type": {"key": "string", "value": "string",
                             "min": 0, "max": "unlimited"}}},
            "isRoot": false},
```
same for pod's LSP's they will get removed when switch goes away so there is no way we leak traffic. OK so based on what we decided for d/s features... we are good here.

Reply:
Oh, I wasn't aware we were removing the switch in that case. Perhaps a comment here would help other people unaware of this since we depend on it.

Reply:
so the latest revision should have a comment that clarifies this @jcaamano ! PTAL